### PR TITLE
Added header and extended setup

### PR DESCRIPTION
**Pull request by Gr1ngoire**
Added Header, some common components and extended initial setup

image

### CODE REVIEW

File: src/components/common/button/button.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    width?: number | string;
+    height?: number | string;
+};
+
+const Button: FC<Props> = ({ name, onClick, width, height }) => (
+    <button
+        className={styles.button}
+        name={name}
+        onClick={onClick}
+        style={{ width, height }}
```
Comment (HoolyDuck) [inline src/components/common/button/button.tsx:16]: is this necessary? maybe just add a classname property and combine it with button sty;e

File: src/components/common/header/components/header-menu/header-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { FC } from "react";
+import Button from "../../../button/button";
+
+import styles from "./styles.module.scss";
+
+const HeaderMenu: FC = () => (
+    <div className={styles.headerMenu}>
+        <Button
+            name={"Profile"}
+            onClick={() => console.log("Go to profile")}
```
Comment (HoolyDuck) [inline src/components/common/header/components/header-menu/header-menu.tsx:10]: I think you can make navigation already

File: src/components/common/header/components/header-menu/header-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            width={100}
+            height={30}
+        />
+        <Button
+            name={"Logout"}
+            onClick={() => console.log("Logout")}
```
Comment (HoolyDuck) [inline src/components/common/header/components/header-menu/header-menu.tsx:16]: same here

File: src/components/common/button/button.tsx (hunk 1 above)
Comment (HoolyDuck) [reply src/components/common/button/button.tsx:16]: also fixed height/width doesn't seem to be a good practice)

File: src/components/common/header/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.header {
+    padding: 0 150px;
+    position: sticky;
+    display: flex;
+    justify-content: space-between;
+    background: black;
+}
+
+.header__textLogo {
+    color: #ffffff;
```
Comment (HoolyDuck) [inline src/components/common/header/styles.module.scss:10]: could use variables but I'll probably going to fix it myself

File: src/components/common/header/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    font-weight: 1000;
+    font-family: "Nosifer", sans-serif;
```
Comment (HoolyDuck) [inline src/components/common/header/styles.module.scss:12]: should create global styles for things like that to not write it everywhere

File: src/components/common/header/components/header-menu/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,6 @@
+.headerMenu {
+    height: 75px;
```
Comment (HoolyDuck) [inline src/components/common/header/components/header-menu/styles.module.scss:2]: sussy fixed height

File: src/components/common/header/header.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        <div className={styles.header}>
+            <h1 className={styles.header__textLogo}>
+                <span className={styles.header_textLogo_coloredFragment}>
+                    PURPLE
+                </span>{" "}
+                MARKET
+            </h1>
+            <div className={styles.header__userData}>
+                <p className={styles.header__username}>DoopSnogg</p>
+                <Image
```
Comment (HoolyDuck) [inline src/components/common/header/header.tsx:19]: are we going to add avatars?)

File: src/components/common/image/image.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    preset: PresetName;
+    onClick?: () => void;
+};
+
+const presetNameToPreset: Record<PresetName, CSSProperties> = {
+    [ImagePreset.AVATAR]: avatarPreset,
+    [ImagePreset.COMMON]: commonPreset,
+};
+
+const Image: FC<Props> = ({ src, preset, onClick }) => (
```
Comment (HoolyDuck) [inline src/components/common/image/image.tsx:32]: very sussy and complicated component, don't even think its necessary and presets can be replaced with simple classnames

**Review by HoolyDuck — CHANGES_REQUESTED** [review]

File: src/components/common/header/styles.module.scss (hunk 5 above)
Comment (HoolyDuck) [reply src/components/common/header/styles.module.scss:12]: going to create them myself

**Comment by HoolyDuck** [comment]
you're also merging to main instead of dev

File: src/components/common/image/image.tsx (hunk 8 above)
Comment (Gr1ngoire) [reply src/components/common/image/image.tsx:32]: Okey, agreed, I was boilerplating configs

File: src/components/common/header/header.tsx (hunk 7 above)
Comment (Gr1ngoire) [reply src/components/common/header/header.tsx:19]: Your Figma contained it :)

**Pull request merged**